libsolidity/formal/SMTEncoder.h

```diff
@@ -199,6 +202,10 @@ class SMTEncoder: public ASTConstVisitor
 	void visitFunctionIdentifier(Identifier const& _identifier);
 	void visitPublicGetter(FunctionCall const& _funCall);
 
+	/// @returns true if @param _contract is set for analysis in the settings
+	/// and it is not abstract.
+	bool shouldAnalyze(ContractDefinition const& _contract) const;
```

Comment:
So, if abstract contract is in a list, we will skip it as most derived. 
I think we should add that to documentation as well, as I think it is missing.

Reply:
Also, should we add a warning that since contract in abstract it will be skipped as deployed contract in model checker?

Reply:
I agree with adding in the docs, but I think a warning would be too much, since it's not really a problem, and it's the correct behavior to not analyze an abstract contract as the most derived